### ISSUE TITLE
Nufft naming conventions

### ISSUE THREAD

**Issue by janden** [ComputationalCryoEM/ASPIRE-Python#223]
There are a number of issues here that should probably be addressed to reduce confusion:

- [x] NFFT vs. NUFFT. Module is called `aspire.nfft`, but contains functions `nufft3` and `anufft3`. For consistency, let's stick with NUFFT as the acronym overall.
- [x] What's `3` in `nufft3` and `anufft3`? Presumably they work *up to* three dimensions, but it's confusing. Let's just have `nufft` and `anufft`.
- [x] Plan class names. `PyNfftPlan` makes sense, but `FINufftPlan` and `cuFINufftPlan` starts getting ugly. Either we follow the convention for acronyms elsewhere in ASPIRE (like `FFBBasis3D`), which would give `FINUFFTPlan` and `CUFINUFFTPlan`, but that is not great either. A compromise might be to just capitalize first letters in acronyms, so we'd have `FinufftPlan` and `CufinufftPlan`. In that case, we'd also need to go and change the others (like `FFBBasis3D`, which would be `FfbBasis3D` or `FastFbBasis3D`).

**Comment by garrettwrong** [comment]
Sure, I can clean these up in the cufinufft PR if you haven't done review already. (Otherwise I'll just do it in another PR).

**Comment by garrettwrong** [comment]
Actually, looking at how many lines this will changes, lets do it after #189

**Comment by janden** [comment]
Yeah that's why I figured I'd make an issue instead of comment on the PR.

**Comment by garrettwrong** [comment]
@janden , I have changed the items listed for `nufft` in #230 .  Before I change other class names, I wanted to bring to attention a rough list of classes that might be affected by this change, and also confirm we want to do this now (as in I'll change them this morning and part of PR 230). 

Some of the classes, ones in the `em_classavg` dir I might ignore, since that code isn't really part of the package afaict.

```
BN:
BNMatrix:
BNMatrixTestCase
CommLineEV
CommLineGCAR
CommLineLUD
CommLineSDP
CommLineSync
CTFFilter
EM:
EMClass2D
EMReconst
FBBasis2D
FBBasis2DTestCase
FBBasis3D
FBBasis3DTestCase
FBBasisImage
FBBasisVolume
FFBBasis2D
FFBBasis2DTestCase
FFBBasis3D
FFBBasis3DTestCase
FPSWF3D
FPSWFBasis2D
FPSWFBasis2DTestCase
MRAClass2D
PSWF2D
PSWF3D
PSWFBasis2D
PSWFBasis2DTestCase
RadialCTFFilter
RFAClass2D
RIRClass2D
```

**Comment by junchaoxia** [comment]
I just want to make a suggestion as I did in issue # 227. Most of code under src/aspire/aspire comes from the original Python code and has not be extracted and merged into the new framework except preprocess functions, PSWF2D basis, and orientation estimation.  I suggest to keep them as is and try to avoid making changes since there are no systematic unit tests to ensure the changes are in right way.

**Comment by janden** [comment]
> Most of code under src/aspire/aspire comes from the original Python code and has not be extracted and merged into the new framework except preprocess functions, PSWF2D basis, and orientation estimation.

Yes, we are aware of that. From what I can tell, the classes that @garrettwrong is listing belong to the new framework and have tests (to a large extent).

Regarding the other class names, we can leave that for now since it'll likely touch too much of the code (and we have ongoing PRs). Let's just do `FinufftPlan` and `CufinufftPlan` for now since those changes are more local. I'll make a separate issue about class naming conventions in general.

**Comment by garrettwrong** [comment]
Sure, that is was I meant by `Some of the classes, ones in the em_classavg dir I might ignore, since that code isn't really part of the package afaict.` .. that I was going to ignore changing them.

Yes, I was considering this would cause some trouble atm. Issue sounds good. Can wait for a good time.

**Comment by junchaoxia** [comment]
Thanks. I also suggest trying to avoid creating new issues related to src/aspire/aspire since we need to extract and put new framework first. We know there are many inconsistencies and it will take much longer time to merge. Let focus on code already in the new framework.

**Comment by janden** [comment]
> I also suggest trying to avoid creating new issues related to src/aspire/aspire since we need to extract and put new framework first.

Again, no one was suggesting we touch those.

**Comment by junchaoxia** [comment]
@janden Sorry for misleading. I tried to put a reminder to all of other since we are discussing it and I also noticed some commits applying changes to files under src/aspire/aspire probably for global consistency. They should be OK but unexpected accidents might happen if unit tests are not ready.
